Cargo.toml

```diff
@@ -206,7 +206,7 @@ opendal = {version = "0.45", default-features = false, features = ["native-tls",
 
 # Tower
 tower = { version = "0.4.13", default-features = false, features = ["buffer", "limit", "retry", "timeout", "util", "balance", "discover"] }
-tower-http = { version = "0.4.4", default-features = false, features = ["decompression-gzip", "trace"]}
+tower-http = { version = "0.4.4", default-features = false, features = ["compression-full", "decompression-gzip", "trace"]}
```

Comment:
I noticed some good discussion in https://github.com/vectordotdev/vector/issues/19962 on if we want to go with an existing library vs writing the implementation ourselves. My two cents is that its simpler in terms of maintenance to use the `tower-http` implementation whether or not we expect to utilize its full feature set. I could see an argument for writing the implementation from scratch if the `tower-http` library and `DecompressionLayer` weren't already used elsewhere in the codebase.

Reply:
Another thing to note is the `compression-gzip` feature can be enabled rather than `compression-full` if we're certain we won't use `br`, `deflate`, `zstd`